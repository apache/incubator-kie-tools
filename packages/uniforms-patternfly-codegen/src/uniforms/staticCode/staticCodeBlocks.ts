/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import dateFunctions from "!!raw-loader!../../resources/dateFunctions.txt";
import timeFunctions from "!!raw-loader!../../resources/timeFunctions.txt";
import selectFunctions from "!!raw-loader!../../resources/selectFunctions.txt";
import multipleSelectFunctions from "!!raw-loader!../../resources/multipleSelectFunctions.txt";
import checkboxGroupFunctions from "!!raw-loader!../../resources/checkboxGroupFunctions.txt";

export const DATE_FUNCTIONS = "date_functions";
export const TIME_FUNCTIONS = "time_functions";
export const SELECT_FUNCTIONS = "select_functions";
export const MULTIPLE_SELECT_FUNCTIONS = "multiple_select_functions";
export const CHECKBOX_GROUP_FUNCTIONS = "checkbox_group_functions";

const _staticBlocks: Map<string, string> = new Map<string, string>();

_staticBlocks.set(DATE_FUNCTIONS, dateFunctions);
_staticBlocks.set(TIME_FUNCTIONS, timeFunctions);
_staticBlocks.set(SELECT_FUNCTIONS, selectFunctions);
_staticBlocks.set(MULTIPLE_SELECT_FUNCTIONS, multipleSelectFunctions);
_staticBlocks.set(CHECKBOX_GROUP_FUNCTIONS, checkboxGroupFunctions);

export const getStaticCodeBlock = (blockName: string): string | undefined => {
  return _staticBlocks.get(blockName);
};
