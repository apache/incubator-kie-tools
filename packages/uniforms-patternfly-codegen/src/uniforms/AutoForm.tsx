/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { Bridge } from "uniforms";
import union from "lodash/union";
import escape from "lodash/escape";
import * as prettier from "prettier";
import { FormElement } from "../api";
import { NS_SEPARATOR } from "./utils/Utils";
import { renderFormInputs } from "./rendering/RenderingUtils";
import { getStaticCodeBlock } from "./staticCode/staticCodeBlocks";

export type AutoFormProps = {
  id: string;
  disabled?: boolean;
  placeholder?: boolean;
  schema: Bridge;
};

const AutoForm: React.FC<AutoFormProps> = (props) => {
  const inputs: FormElement<any>[] = renderFormInputs(props.schema);

  let pfImports: string[] = [];
  let reactImports: string[] = [];
  let staticCodeArray: string[] = [];

  inputs.forEach((input) => {
    pfImports = union(pfImports, input.pfImports);
    reactImports = union(reactImports, input.reactImports);
    staticCodeArray = union(staticCodeArray, input.requiredCode);
  });

  const formName = `Form${props.id ? `${NS_SEPARATOR}${props.id}` : ""}`;
  const hooks = inputs.map((input) => input.stateCode).join("\n");
  const elements = inputs.map((input) => input.jsxCode).join("\n");
  const staticCodeStr: string = staticCodeArray.map((id) => getStaticCodeBlock(id)).join("\n");

  const formTemplate = `import React, { ${reactImports.join(", ")} }  from "react";
    import { ${pfImports.join(", ")} } from "@patternfly/react-core";
    
    const ${formName}: React.FC<any> = ( props:any ) => {
      ${hooks}
      
      ${staticCodeStr}
      
      return (
        <div className={'pf-c-form'}>
          ${elements}        
        </div>
      )    
    }
    
    export default ${formName};`;

  const formattedFormTemplate = prettier.format(formTemplate, {
    parser: "typescript",
    singleQuote: true,
    jsxSingleQuote: true,
    jsxBracketSameLine: true,
    tabWidth: 2,
  });
  return <>{escape(formattedFormTemplate)}</>;
};

export default AutoForm;
