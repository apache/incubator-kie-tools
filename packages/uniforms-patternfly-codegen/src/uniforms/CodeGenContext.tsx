/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { createContext, useContext } from "react";

import { Bridge, context, Context, randomIds } from "uniforms/cjs";

import { FormElement } from "../api";

export interface CodeGenContext {
  rendered: FormElement[];
}

export const codeGenContext = createContext<CodeGenContext | null>(null);

export const useCodegenContext = (): CodeGenContext | null => {
  return useContext<CodeGenContext | null>(codeGenContext);
};

export const useAddFormElementToContext = (formElement: FormElement): void => {
  const ctx = useCodegenContext();
  if (!ctx) {
    throw new Error(`'useAddFormElementToContext' should be called within a 'codegenContext'`);
  }
  ctx.rendered.push(formElement);
};

export interface ProviderProps {
  schema: Bridge;
  codegenCtx: CodeGenContext;
  uniformsCtx?: Context<any>;
  children: JSX.Element;
}

export const CodeGenContextProvider: React.FC<ProviderProps> = (props) => {
  const ctx: Context<any> = props.uniformsCtx || {
    changed: false,
    changedMap: {},
    error: false,
    model: {},
    name: [],
    onChange: (key, value) => {},
    onSubmit: (event) => {},
    randomId: randomIds(),
    schema: props.schema,
    state: {
      readOnly: false,
      disabled: false,
      label: true,
      placeholder: true,
      showInlineError: true,
    },
    submitted: false,
    submitting: false,
    validating: false,
  };

  return (
    <codeGenContext.Provider value={props.codegenCtx}>
      <context.Provider value={ctx}>{props.children}</context.Provider>
    </codeGenContext.Provider>
  );
};
