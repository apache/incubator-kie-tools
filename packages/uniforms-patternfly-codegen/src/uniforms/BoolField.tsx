/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { CheckboxProps } from "@patternfly/react-core";
import { connectField } from "uniforms";
import { useCodegenContext } from "./CodeGenContext";
import { InputReference } from "../api";
import { buildDefaultInputElement, getInputReference, renderField } from "./utils/Utils";

export type BoolFieldProps = {
  appearance?: "checkbox" | "switch";
  label: string;
  disabled: boolean;
  required: boolean;
  onChange: (value?: string) => void;
} & Omit<CheckboxProps, "isDisabled">;

const Bool: React.FC<BoolFieldProps> = (props: BoolFieldProps) => {
  const codegenContext = useCodegenContext();

  const ref: InputReference = getInputReference(props.name);

  const jsxCode = `<Checkbox
      isChecked={props.value || false}
      isDisabled={${props.disabled || "false"}}
      id="${props.id}"
      name="${props.name}"
      label="${props.label}"
      onChange={${ref.stateSetter}}
    />`;

  const element = buildDefaultInputElement(["Checkbox"], jsxCode, ref, "boolean", props);
  codegenContext.rendered.push(element);

  return renderField(element);
};

export default connectField(Bool);
