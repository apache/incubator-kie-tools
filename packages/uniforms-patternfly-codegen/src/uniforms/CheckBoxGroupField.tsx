/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { connectField, HTMLFieldProps } from "uniforms/es5";
import { buildDefaultInputElement, getInputReference, renderField } from "./utils/Utils";
import { FormInput, InputReference } from "../api";
import { useAddFormElementToContext } from "./CodeGenContext";
import { CHECKBOX_GROUP_FUNCTIONS } from "./staticCode/staticCodeBlocks";

export type CheckBoxGroupProps = HTMLFieldProps<
  string[],
  HTMLDivElement,
  {
    name: string;
    label: string;
    allowedValues?: string[];
    required: boolean;
    transform?(value: string): string;
  }
>;

const CheckBoxGroup: React.FC<CheckBoxGroupProps> = (props: CheckBoxGroupProps) => {
  const ref: InputReference = getInputReference(props.name);

  const jsxCode = props.allowedValues
    ?.map((value) => {
      return `<Checkbox key={'${props.id}-${value}'} id={'${props.id}-${value}'} name={'${props.name}'} aria-label={'${
        props.name
      }'}
               label={'${props.transform ? props.transform(value) : value}'} 
               isDisabled={${props.disabled || false}} 
               isChecked={${ref.stateName}.indexOf('${value}') != -1}
               onChange={() => handleCheckboxGroupChange('${value}', ${ref.stateName}, ${ref.stateSetter})}
               value={'${value}'}/>`;
    })
    .join("\n");

  const element: FormInput = buildDefaultInputElement({
    pfImports: ["Checkbox"],
    inputJsxCode: jsxCode || "",
    ref: ref,
    dataType: "string[]",
    defaultValue: "[]",
    requiredCode: [CHECKBOX_GROUP_FUNCTIONS],
    wrapper: {
      id: props.id,
      label: props.label,
      required: props.required,
    },
  });

  useAddFormElementToContext(element);

  return renderField(element);
};

export default connectField(CheckBoxGroup);
