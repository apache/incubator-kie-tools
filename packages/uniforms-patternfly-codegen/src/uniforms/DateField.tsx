/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { connectField, HTMLFieldProps } from "uniforms/cjs";

import { FormInput, InputReference } from "../api";
import { buildDefaultInputElement, getInputReference, renderField } from "./utils/Utils";
import { useAddFormElementToContext } from "./CodeGenContext";
import { DATE_FUNCTIONS, TIME_FUNCTIONS } from "./staticCode/staticCodeBlocks";
import { DATE } from "./utils/dataTypes";

export type DateFieldProps = HTMLFieldProps<
  Date,
  HTMLDivElement,
  {
    name: string;
    label: string;
    required: boolean;
    max?: Date;
    min?: Date;
  }
>;

const Date: React.FC<DateFieldProps> = (props: DateFieldProps) => {
  const ref: InputReference = getInputReference(props.name, DATE);

  const pfImports = ["DatePicker", "Flex", "FlexItem", "InputGroup", "TimePicker"];

  const jsxCode = `<Flex
    direction={{ default: 'column' }}
    id={'${props.id}'}
  >
    <FlexItem>
      <InputGroup style={{ background: 'transparent' }}>
        <DatePicker
          id={'date-picker-${props.id}'}
          isDisabled={${props.disabled || false}}
          name={'${props.name}'}
          onChange={newDate => onDateChange(newDate, ${ref.stateSetter},  ${ref.stateName})}
          value={parseDate(${ref.stateName})}
        />
        <TimePicker
          id={'time-picker-${props.id}'}
          isDisabled={${props.disabled || false}}
          name={'${props.name}'}
          onChange={(time, hours?, minutes?) => onTimeChange(time, ${ref.stateSetter}, ${
    ref.stateName
  }, hours, minutes)}
          style={{ width: '120px' }}
          time={parseTime(${ref.stateName})}
        />
      </InputGroup>
    </FlexItem>
  </Flex>`;

  const element: FormInput = buildDefaultInputElement({
    pfImports,
    inputJsxCode: jsxCode,
    ref,
    requiredCode: [DATE_FUNCTIONS, TIME_FUNCTIONS],
    wrapper: {
      id: props.id,
      label: props.label,
      required: props.required,
    },
    disabled: props.disabled,
  });

  useAddFormElementToContext(element);

  return renderField(element);
};

export default connectField(Date);
