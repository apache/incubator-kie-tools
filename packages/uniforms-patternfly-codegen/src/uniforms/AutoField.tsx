/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { createAutoField } from "uniforms/cjs/createAutoField";

import TextField from "./TextField";
import BoolField from "./BoolField";
import NumField from "./NumField";
import NestField from "./NestField";
import DateField from "./DateField";
import RadioField from "./RadioField";
import SelectField from "./SelectField";
import CheckBoxGroupField from "./CheckBoxGroupField";
import UnsupportedField from "./UnsupportedField";

export type AutoFieldProps = Parameters<typeof AutoField>[0];

const AutoField = createAutoField((props) => {
  if (props.allowedValues) {
    if (props.checkboxes) {
      return props.fieldType !== Array ? RadioField : CheckBoxGroupField;
    }
    return SelectField;
  }

  switch (props.fieldType) {
    /*
    TODO: implement array support
    case Array:
      return  ListField;*/
    case Boolean:
      return BoolField;
    case Date:
      return DateField;
    case Number:
      return NumField;
    case Object:
      return NestField;
    case String:
      return TextField;
  }

  console.log(`Unsupported field type: ${props.fieldType}`);
  return UnsupportedField;
});

export default AutoField;
