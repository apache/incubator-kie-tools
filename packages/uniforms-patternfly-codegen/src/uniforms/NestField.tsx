/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext } from "react";
import { connectField, context, HTMLFieldProps } from "uniforms/es5";
import { renderNestedInputFragmentWithContext } from "./rendering/RenderingUtils";
import { renderField } from "./utils/Utils";
import { InputReference, InputsContainer } from "../api";
import { codeGenContext } from "./CodeGenContext";
import { union } from "lodash";

export type NestFieldProps = HTMLFieldProps<object, HTMLDivElement, { itemProps?: object }>;

const Nest: React.FunctionComponent<NestFieldProps> = ({
  id,
  children,
  error,
  errorMessage,
  fields,
  itemProps,
  label,
  name,
  showInlineError,
  disabled,
  ...props
}: NestFieldProps) => {
  const uniformsContext = useContext(context);
  const codegenCtx = useContext(codeGenContext);

  const nestedRefs: InputReference[] = [];
  const nestedStates: string[] = [];
  const nestedJsx: string[] = [];

  let pfImports: string[] = ["Card", "CardBody"];
  let reactImports: string[] = [];
  let requiredCode: string[] = [];

  if (fields) {
    fields.forEach((field) => {
      const renderedInput = renderNestedInputFragmentWithContext(uniformsContext, field, itemProps, disabled);

      if (renderedInput) {
        nestedStates.push(renderedInput.stateCode);
        nestedJsx.push(renderedInput.jsxCode);
        nestedRefs.push(renderedInput.ref);
        pfImports = union(pfImports, renderedInput.pfImports);
        reactImports = union(reactImports, renderedInput.reactImports);
        if (renderedInput.requiredCode) {
          requiredCode = union(requiredCode, renderedInput.requiredCode);
        }
      } else {
        console.log(`Cannnot render form field for: '${field}'`);
      }
    });
  }

  const bodyLabel = label ? `<label><b>${label}</b></label>` : "";

  const stateCode = nestedStates.join("\n");
  const jsxCode = `<Card>
          <CardBody className="pf-c-form">
          ${bodyLabel}
          ${nestedJsx.join("\n")}
          </CardBody></Card>`;

  const rendered: InputsContainer = {
    pfImports,
    reactImports,
    requiredCode: requiredCode,
    stateCode,
    jsxCode,
    ref: nestedRefs,
  };

  codegenCtx?.rendered.push(rendered);

  return renderField(rendered);
};

export default connectField(Nest);
