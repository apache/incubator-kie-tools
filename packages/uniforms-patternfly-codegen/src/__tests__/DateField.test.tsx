/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { renderField } from "./_render";
import { DATE_FUNCTIONS, TIME_FUNCTIONS } from "../uniforms/staticCode/staticCodeBlocks";
import { DateField } from "../uniforms";

const schema = {
  birthday: { type: Date },
};

describe("<DateField> tests", () => {
  it("<DateField> - rendering", () => {
    const props = {
      id: "id",
      label: "Birthday",
      name: "birthday",
      disabled: false,
      onChange: jest.fn(),
    };

    const { container, formElement } = renderField(DateField, props, schema);

    expect(container).toMatchSnapshot();

    expect(formElement.reactImports).toContain("useState");
    expect(formElement.pfImports).toContain("FormGroup");
    expect(formElement.pfImports).toContain("DatePicker");
    expect(formElement.pfImports).toContain("TimePicker");
    expect(formElement.requiredCode).not.toBeUndefined();
    expect(formElement.requiredCode).toHaveLength(2);
    expect(formElement.requiredCode).toContain(DATE_FUNCTIONS);
    expect(formElement.requiredCode).toContain(TIME_FUNCTIONS);
    expect(formElement.ref.binding).toBe(props.name);
    expect(formElement.ref.stateName).toBe(props.name);
    expect(formElement.ref.stateSetter).toBe(`set__${props.name}`);

    expect(formElement.jsxCode).not.toBeNull();
    expect(formElement.jsxCode).toContain(`label={'${props.label}'}`);
    expect(formElement.jsxCode).toContain(`name={'${props.name}'}`);
    expect(formElement.jsxCode).toContain("isDisabled={false}");
    expect(formElement.jsxCode).toContain(
      `onChange={newDate => onDateChange(newDate, ${formElement.ref.stateSetter},  ${formElement.ref.stateName})}`
    );
    expect(formElement.jsxCode).toContain(`value={parseDate(${formElement.ref.stateName})}`);
    expect(formElement.jsxCode).toContain(`time={parseTime(${formElement.ref.stateName})}`);
    expect(formElement.jsxCode).toContain(
      `onChange={(time, hours?, minutes?) => onTimeChange(time, ${formElement.ref.stateSetter}, ${formElement.ref.stateName}, hours, minutes)}`
    );
    expect(formElement.stateCode).not.toBeNull();
  });

  it("<DateField> - rendering - disabled", () => {
    const props = {
      id: "id",
      label: "Birthday",
      name: "birthday",
      disabled: true,
      onChange: jest.fn(),
    };

    const { container, formElement } = renderField(DateField, props, schema);

    expect(container).toMatchSnapshot();

    expect(formElement.jsxCode).not.toBeNull();
    expect(formElement.jsxCode).toContain(`label={'${props.label}'}`);
    expect(formElement.jsxCode).toContain(`name={'${props.name}'}`);
    expect(formElement.jsxCode).toContain("isDisabled={true}");
  });
});
