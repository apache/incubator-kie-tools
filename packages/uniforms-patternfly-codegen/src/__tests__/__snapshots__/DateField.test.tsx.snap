// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateField> tests <DateField> - rendering - disabled 1`] = `
<div>
  {"ref":{"binding":"birthday","stateName":"birthday","stateSetter":"set__birthday","dataType":{"name":"string"}},"pfImports":["DatePicker","Flex","FlexItem","InputGroup","TimePicker","FormGroup"],"reactImports":["useState"],"requiredCode":["date_functions","time_functions"],"jsxCode":"&lt;FormGroup\\n      fieldId={'id'}\\n      label={'Birthday'}\\n      isRequired={true}\\n    &gt;\\n      &lt;Flex\\n    direction={{ default: 'column' }}\\n    id={'id'}\\n  &gt;\\n    &lt;FlexItem&gt;\\n      &lt;InputGroup style={{ background: 'transparent' }}&gt;\\n        &lt;DatePicker\\n          id={'date-picker-id'}\\n          isDisabled={true}\\n          name={'birthday'}\\n          onChange={newDate =&gt; onDateChange(newDate, set__birthday,  birthday)}\\n          value={parseDate(birthday)}\\n        /&gt;\\n        &lt;TimePicker\\n          id={'time-picker-id'}\\n          isDisabled={true}\\n          name={'birthday'}\\n          onChange={(time, hours?, minutes?) =&gt; onTimeChange(time, set__birthday, birthday, hours, minutes)}\\n          style={{ width: '120px' }}\\n          time={parseTime(birthday)}\\n        /&gt;\\n      &lt;/InputGroup&gt;\\n    &lt;/FlexItem&gt;\\n  &lt;/Flex&gt;\\n    &lt;/FormGroup&gt;","stateCode":"const [ birthday, set__birthday ] = useState&lt;string&gt;();","isReadonly":true}
</div>
`;

exports[`<DateField> tests <DateField> - rendering 1`] = `
<div>
  {"ref":{"binding":"birthday","stateName":"birthday","stateSetter":"set__birthday","dataType":{"name":"string"}},"pfImports":["DatePicker","Flex","FlexItem","InputGroup","TimePicker","FormGroup"],"reactImports":["useState"],"requiredCode":["date_functions","time_functions"],"jsxCode":"&lt;FormGroup\\n      fieldId={'id'}\\n      label={'Birthday'}\\n      isRequired={true}\\n    &gt;\\n      &lt;Flex\\n    direction={{ default: 'column' }}\\n    id={'id'}\\n  &gt;\\n    &lt;FlexItem&gt;\\n      &lt;InputGroup style={{ background: 'transparent' }}&gt;\\n        &lt;DatePicker\\n          id={'date-picker-id'}\\n          isDisabled={false}\\n          name={'birthday'}\\n          onChange={newDate =&gt; onDateChange(newDate, set__birthday,  birthday)}\\n          value={parseDate(birthday)}\\n        /&gt;\\n        &lt;TimePicker\\n          id={'time-picker-id'}\\n          isDisabled={false}\\n          name={'birthday'}\\n          onChange={(time, hours?, minutes?) =&gt; onTimeChange(time, set__birthday, birthday, hours, minutes)}\\n          style={{ width: '120px' }}\\n          time={parseTime(birthday)}\\n        /&gt;\\n      &lt;/InputGroup&gt;\\n    &lt;/FlexItem&gt;\\n  &lt;/Flex&gt;\\n    &lt;/FormGroup&gt;","stateCode":"const [ birthday, set__birthday ] = useState&lt;string&gt;();","isReadonly":false}
</div>
`;
